### PR TITLE
Improve readme

### PR DESCRIPTION
**Pull request by martinda**
I would like to improve the README as presented in this PR. Full transparency, I am working with @nurgul212 and @olivierdagenais on this plugin.

### How the README is improved

I am improving the README by adding a more extensive description of rule construction, rule components, and over all operation. I separately describe built-in conditions, user specified conditions, and user specified actions. I also added a description of the algorithm, so that users can have a better understanding of the plugin operation before they decide to install it and before they try it on their system.

```[tasklist]
### Submitter checklist
- [x] Make sure you are opening from a **topic/feature/bugfix branch** (right side) and not your main branch!
- [x] Ensure that the pull request title represents the desired changelog entry
- [x] Please describe what you did
- [ ] Link to relevant issues in GitHub or Jira
- [ ] Link to relevant pull requests, esp. upstream and downstream changes
- [ ] Ensure you have provided tests - that demonstrates feature works or fixes the issue
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -11,44 +11,104 @@
...
+The actions are applied in the order they are defined.
+
+The actions are:
+
+- [Change build description](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/ChangeBuildDescriptionAction.java)
+- [Delete artifacts](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteArtifactsAction.java)
+- [Delete build](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteBuildAction.java)
+- [Delete log file](./src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/model/actions/DeleteLogFileAction.java) 
+
+If the first action is to delete the build, the other actions will still be applied but have no real effect.
```
Comment (damianszczepanik) [inline README.md:76]: I'm not so sure.
Build deletion is not performed synchronously so I can imagine configuration and runtime execution when first action is to delete the build and second is to keep the build forever. I'm not sure what happens then :-)

File: README.md [hunk 2]
```diff
@@ -11,44 +11,104 @@
...
+               the plugin stops applying rules to this build, and goes to the next iteration of the "for each completed build" loop.
+        3. If one or more optional condition are not met,
+           go to the next completed build (next iteration of the "for each completed build" loop).
+
+__Notes__:
+
+1. If the `matchAtMost` value is set to zero, the rule is effectively disabled.
+   This is useful if the user wants to disable a rule while keeping its other configuration values.
+2. Once the `matchAtMost` value is reached, the rule is effectively disabled and is no longer applied.
+3. It may not make sense to continue to apply rules after a build is deleted, but the plugin handles this case gracefully.
```
Comment (damianszczepanik) [inline README.md:106]: that's the case I've described above - happy you have noticed that

File: README.md [hunk 3]
```diff
@@ -11,44 +11,104 @@
...
+4. Having no condition is a way to unconditionally apply actions to builds, for example to delete all the builds.
```
Comment (damianszczepanik) [inline README.md:107]: Suggest to add here: use it wisely

**Comment by damianszczepanik** [comment]
I'm happy to read your review!

**Review by damianszczepanik — APPROVED** [review]

**Comment by damianszczepanik** [comment]
I believe a few of above valuable information can be copied to https://github.com/jenkinsci/build-history-manager-plugin/tree/master/src/main/resources/pl/damianszczepanik/jenkins/buildhistorymanager/model - prepare PR if you think this is worth to add

**Comment by martinda** [comment]
> I believe a few of above valuable information can be copied to https://github.com/jenkinsci/build-history-manager-plugin/tree/master/src/main/resources/pl/damianszczepanik/jenkins/buildhistorymanager/model - prepare PR if you think this is worth to add

Oh you mean to make similar documentation updates to the jelly and html files where appropriate. Okay I agree, it makes sense.

**Comment by nurgul212** [comment]
we have now included the important information in the corresponding help.html files. Pull request #88 has been sent.

**Pull request merged**